### PR TITLE
feat(editor): Add ability to extract sub-workflows to canvas context menu

### PR DESCRIPTION
**Pull request by CharlieKolb**
## Summary

Add sub-workflow extraction functionality

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-3562/feature-implement-sub-workflow-conversion-core-feature

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/frontend/editor-ui/src/composables/useWorkflowExtraction.ts [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+
+		return {
+			name: newWorkflowName,
+			nodes: [...nodes, ...returnNode, triggerNode],
+			connections: {
+				...newConnections,
+				...startNodeConnection,
+				...endNodeConnection,
+			},
+			settings: { executionOrder: 'v1' },
```
Comment (Cadiac) [inline packages/frontend/editor-ui/src/composables/useWorkflowExtraction.ts:242]: Should we refuse to do the extraction if the workflow we're extracting from is some legacy one not using `v1`?

Comment (Cadiac) [reply packages/frontend/editor-ui/src/composables/useWorkflowExtraction.ts:242]: I have no idea how often those really exist in the wild, but I do know that they are executed slightly differently on the execution engine - if we extract nodes from them into sub workflows is there a risk something changes? 🤔 

Then again this might be an edge case we can happily scope out of the MVP

File: packages/workflow/test/GraphUtils/graph-utils.test.ts [hunk 2]
```diff
@@ -1,323 +0,0 @@
-import {
```
Comment (Cadiac) [inline packages/workflow/test/GraphUtils/graph-utils.test.ts:1]: These were removed, intentional?

**Review by Cadiac — APPROVED** [review]
Awesome work, very clean PR and one of my favorite features! 🚀

**Review by Cadiac — DISMISSED** [review]

**Review by Cadiac — DISMISSED** [review]

**Comment by shortstacked** [comment]
## Workflow Test Results 📊 🔴 **2 Failed,** ⚠️ 3 Warnings, 👍 78 Successful out of 83 total workflows.
*Detail:* Workflows failing: *237*: Workflow contains 1 deleted data. [View full workflow run](https://github.com/n8n-io/n8n/actions/runs/15387057281)

*Tested Ref:* `f5e006c4cb3ef31e91639d504fd2dcb8909218f5` by @Cadiac
### ❌ Failed Tests (2)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 237 | BasicLLMChain:AzureChat | Workflow contains 1 deleted data. |
| 84 | Matrix:Room:create invite kick leave:RoomMember:ge... | Too Many Requests on node Matrix2 |
### ⚠️ Warnings (3)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 35 | Slack:User:getPresence info:UserProfile:get update... | Workflow contains new data that previously did not exist. |
| 257 | Agent:auto-fix:anthropic | Workflow contains new data that previously did not exist. |
| 53 | ConvertKit:CustomField:create getAll update delete... | Workflow contains new data that previously did not exist. |

File: packages/frontend/@n8n/i18n/src/locales/en.json [hunk 3]
```diff
@@ -1521,6 +1521,7 @@
 	"contextMenu.deselectAll": "Clear selection",
 	"contextMenu.tidyUpWorkflow": "Tidy up workflow",
 	"contextMenu.tidyUpSelection": "Tidy up selection",
+	"contextMenu.extract": "Extract node into Sub-workflow | Extract {count} nodes into Sub-workflow",
```
Comment (RobTF9) [inline packages/frontend/@n8n/i18n/src/locales/en.json]: Don't capitalise "sub-workflow"

File: packages/frontend/@n8n/i18n/src/locales/en.json [hunk 4]
```diff
@@ -2368,6 +2369,21 @@
 	"workflowDetails.active": "Active",
 	"workflowDetails.addTag": "Add tag",
 	"workflowDetails.chooseOrCreateATag": "Choose or create a tag",
+	"workflowExtraction.error.failure": "Sub-workflow Extraction failed",
```
Comment (RobTF9) [inline packages/frontend/@n8n/i18n/src/locales/en.json]: "Extraction" should be lowercase

File: packages/frontend/@n8n/i18n/src/locales/en.json [hunk 5]
```diff
@@ -2368,6 +2369,21 @@
...
+	"workflowExtraction.error.selectionGraph.inputEdgeToNonRoot": "Non-input node '{node}' has a connection from outside the selection.",
```
Comment (RobTF9) [inline packages/frontend/@n8n/i18n/src/locales/en.json]: "Non-input node '{node}' has a connection **from** a node outside the current selection."

File: packages/frontend/@n8n/i18n/src/locales/en.json [hunk 6]
```diff
@@ -2368,6 +2369,21 @@
...
+	"workflowExtraction.error.selectionGraph.outputEdgeFromNonLeaf": "Non-output node '{node}' has a connection to outside the selection.",
```
Comment (RobTF9) [inline packages/frontend/@n8n/i18n/src/locales/en.json]: "Non-output node '{node}' has a connection **to** a node outside the current selction."

Comment (CharlieKolb) [reply packages/frontend/@n8n/i18n/src/locales/en.json]: To confirm, also drop the `'` around the node name? This reads a bit confusing sometimes, e.g. `Non-output node add entry in google sheets has a connection...`

Comment (RobTF9) [reply packages/frontend/@n8n/i18n/src/locales/en.json]: No, my mistake leave the `''`

**Review by Cadiac — DISMISSED** [review]

**Comment by shortstacked** [comment]
## Workflow Test Results 📊 🔴 **2 Failed,** ⚠️ 3 Warnings, 👍 78 Successful out of 83 total workflows.
*Detail:* Workflows failing: *237*: Workflow contains 1 deleted data. [View full workflow run](https://github.com/n8n-io/n8n/actions/runs/15387916264)

*Tested Ref:* `ecd2326d68719639b00dcd7d9f658c3e4032b048` by @Cadiac
### ❌ Failed Tests (2)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 237 | BasicLLMChain:AzureChat | Workflow contains 1 deleted data. |
| 258 | Agent:auto-fix:openai | Workflow contains 2 deleted data. |
### ⚠️ Warnings (3)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 35 | Slack:User:getPresence info:UserProfile:get update... | Workflow contains new data that previously did not exist. |
| 53 | ConvertKit:CustomField:create getAll update delete... | Workflow contains new data that previously did not exist. |
| 257 | Agent:auto-fix:anthropic | Workflow contains new data that previously did not exist. |

**Review by Cadiac — APPROVED** [review]

**Comment by shortstacked** [comment]
## Workflow Test Results 📊 🔴 **1 Failed,** ⚠️ 3 Warnings, 👍 79 Successful out of 83 total workflows.
*Detail:* Workflows failing: *237*: Workflow contains 1 deleted data. [View full workflow run](https://github.com/n8n-io/n8n/actions/runs/15389278173)

*Tested Ref:* `fffb773166909948e1b3933b0e9d92d76dc155bb` by @Cadiac
### ❌ Failed Tests (1)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 237 | BasicLLMChain:AzureChat | Workflow contains 1 deleted data. |
### ⚠️ Warnings (3)
| Workflow ID | Workflow Name | Reason |
|-------------|---------------|--------|
| 35 | Slack:User:getPresence info:UserProfile:get update... | Workflow contains new data that previously did not exist. |
| 257 | Agent:auto-fix:anthropic | Workflow contains new data that previously did not exist. |
| 53 | ConvertKit:CustomField:create getAll update delete... | Workflow contains new data that previously did not exist. |

**Comment by janober** [comment]
Got released with [`n8n@1.97.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.97.0)
